src/IonTextReader.ts

```diff
@@ -92,8 +92,23 @@ export class TextReader implements Reader {
         if(this.depth() > 0) return false;
         if(input.length < 6 || this.annotations().length > 0) return false;
         let prefix = "$ion_";
-        for(let i = 0; i < prefix.length; i++){
+        let i = 0;
+
+        while(i < prefix.length){
             if(prefix.charAt(i) !== input.charAt(i)) return false;
+            i++;
+        }
+        let tempChar;
```

Comment:
- define this var within the loop, as it has no value outside the loop
- consider renaming it to 'ch'

Reply:
done